src/ui/bounding_widget_2d.h

```diff
@@ -39,6 +39,9 @@ class Bounding_Polygon_Widget {
     float split_point_size = 7.f;
     float center_point_size = 12.f;
     float selected_center_point_size = 14.f;
+#ifdef __APPLE__
+    float macos_widget_scaling_factor = 4.5f; // Scaling factor for widget window range on macOS
```

Comment:
can you explain why it's set to 4.5. At some point, somebody is going to read this code and have to figure out why this is the magic number that works. Explain the different conversions and what's going on. This comment should be detailed.